pkg/chains/objects/objects.go

```diff
@@ -40,14 +43,158 @@ type Object interface {
 	runtime.Object
 }
 
-// Result is a generic key value store containing the results
+// GenericResult is a generic key value store containing the results
 // of Tekton operations. (eg. PipelineRun and TaskRun results)
-type Result struct {
+type GenericResult interface {
```

Comment:
I think it's okay to upgrade the existing Result struct to v1. We're already making a breaking library change here, and these types are identical.

Reply:
I'm fine with making the change you suggest.  The general idea in the PR here was to make it so that there was no versioned objects in Tekton Object interface (other than `Object`):
```
// Tekton object is an extended Kubernetes object with operations specific
// to Tekton objects.
type TektonObject interface {
	Object
	GetGVK() string
	GetKindName() string
	GetObject() interface{}
	GetLatestAnnotations(ctx context.Context, clientSet versioned.Interface) (map[string]string, error)
	Patch(ctx context.Context, clientSet versioned.Interface, patchBytes []byte) error
	GetResults() []GenericResult
	GetProvenance() GenericProvenance
	GetServiceAccountName() string
	GetPullSecrets() []string
	IsDone() bool
	IsSuccessful() bool
	SupportsTaskRunArtifact() bool
	SupportsPipelineRunArtifact() bool
	SupportsOCIArtifact() bool
	GetRemoteProvenance() GenericProvenance
	IsRemote() bool
}
```

If that goal is not necessary, I am happy to make this interface rely on a specific version of the Result interface in (this case `v1`) and change it over time as well

Reply:
I've updated the code here to use the v1 Result and removed the GenericResult.  For v1beta1 the v1beta1 Result info they are converted to v1 Result in `GetResults`